ci: move post processing to the CI script

nashif opened the pull request:
Move post processing to script and remove duplicatated code.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
ci: move pos processing to the CI script

Move post processing to script and remove duplicatated code.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

aescolar commented on scripts/ci/run_ci.sh:
```diff
@@ -54,22 +66,90 @@ while getopts ":pm:b:r:M:" opt; do
 
 DOC_MATRIX=${MATRIX_BUILDS}
-if [ -z "$BRANCH" ]; then
+if [ -n "$MAIN_CI" -a -z "$BRANCH" ]; then
```
It feels like wrapping these 3 ifs in a `if [ -n  "$MATRIX_CI" ]` instead of repeating it 3 times would have been cleaner/more obvious to the reader.

aescolar commented on scripts/ci/run_ci.sh:
```diff
@@ -54,22 +66,90 @@ while getopts ":pm:b:r:M:" opt; do
-	exit
 fi
 
+function handle_compiler_cache() {
```
dont you want to move this function a bit lower so its next to the caller?

aescolar commented on scripts/ci/run_ci.sh:
```diff
@@ -100,17 +180,17 @@ function run_bsim_bt_tests() {
+	echo "- Verify commit message, coding style, doc build";
 }
 
 function get_tests_to_run() {
```
by the way the indentation is still inconsistent (spaces vs tabs)

aescolar reviewed (dismissed):
It seems ok.
Silly thing: In the commit msg: "duplicatated" (And the pos=> post thingy)

The rest is just about bike sheds

(I removed the approval due to the missing codecov report)

aescolar commented:
Why did the codecov results not get posted? (they do seem uploaded https://codecov.io/gh/zephyrproject-rtos/zephyr/tree/5cec2fb5e1daeaa6c2acb43a6aac2eb15f8d1f7d )

aescolar commented:
@nashif Trying to guess why codecov did not upload things makes me think it would be nicer running the script verbose (`set -v`) always. What do you think?

aescolar commented:
recheck

aescolar commented on scripts/ci/run_ci.sh:
```diff
@@ -16,6 +16,9 @@
 # The script can be run locally using for exmaple:
 # ./scripts/ci/run_ci.sh -b master -r upstream  -p
+
+set -v
```
Ok.. `set -v` is just annoying printing all functions and the options switch.. sorry for that.
`set -x` seems to be much more useful.
Note that zephyr-env.sh also gets verbose. If that's what you wanted, then its fine, but if not, you may want to move the set -x just below the call to `source zephyr-env.sh`.

aescolar reviewed (commented):
Apart from my own bad proposal I'm happy with this.
I still do not understand why codecov did not post the results in this PR, but everything seems uploaded properly.

nashif committed:
ci: define BSIM_* variables in ci script directly

Rather than pass them as options, just set it directly in the CI script.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
ci: remove sanitycheck options from .shippable

They are now defined in CI script directly.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
ci: pass pull request number to CI script

For further processing, pass pull request number to the CI script.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
ci: use set -x for verbosity

set -v was too verbose, so change to -x and remove extra sourcing of
zephyr-env.sh.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

aescolar reviewed (approved):
Nice
